apps/shopify/src/skuResolvers.js

```diff
@@ -26,37 +29,41 @@ export async function makeShopifyClient(config) {
   const { storefrontAccessToken, apiEndpoint } = config;
 
   return Client.buildClient({
-    domain: apiEndpoint,
+    domain: removeHttpsAndTrailingSlash(apiEndpoint),
     storefrontAccessToken,
     apiVersion: SHOPIFY_API_VERSION,
   });
 }
 
 /**
  * Fetches the collection previews for the collections selected by the user.
- *
- * Note: currently there is no way to fetch multiple collections by id
- * so we use fetchAll instead and then filter on the client. Besides the obvious disadvantage,
- * this could also fail if there are mo collections in the stroe than the pagination limit
  */
 export const fetchCollectionPreviews = async (skus, config) => {
   if (!skus.length) {
     return [];
   }
 
   const validIds = filterAndDecodeValidIds(skus, 'Collection');
-
   const shopifyClient = await makeShopifyClient(config);
 
-  const response = await shopifyClient.collection.fetchAll(250);
+  const response = await shopifyClient.collection.fetchAll(SHOPIFY_ENTITY_LIMIT);
+  if (response.length > 0) {
+    while (response[response.length - 1].hasNextPage) {
```

Comment:
Interesting. Is `.hasNextPage` a field that Shopify gives us? Pretty cool if so.

Reply:
Yup! It's a field attached to the the response nodes! It's sick